`test_apply_GS_hra_conv2d_inference` fails with FileNotFoundError due to datasets>=4.0 incompatibility with `trust_remote_code=True`

### System Info

**Test Failure:**
`tests/test_common_gpu.py::PeftGPUCommonTests::test_apply_GS_hra_conv2d_inference`

**Error:**
```
FileNotFoundError: [Errno 2] No such file or directory: '/root/.cache/huggingface/hub/datasets--huggingface--cats-image/snapshots/4613f5f1d3642cc2d56ffdf1b58c9d0f912cdc1f/cats_image.jpeg'
```

**Root Cause:**
The datasets library version >4.0 no longer supports the trust_remote_code=True parameter for the load_dataset API (referenced at [L292 ](https://github.com/huggingface/peft/blob/main/tests/testing_utils.py#L292) in the test code). This prevents the cats-image dataset from being downloaded properly during test execution.

This is a device-agnostic issue - it appears to work on other devices only because the image was cached from previous runs. On clean environments or other devices, the test consistently fails.

**Workaround:**
1. Downgrade the datasets library to version <4.0
2. Pre-download the image before running tests

@BenjaminBossan  - Could you please take a look at this issue ?

Environment:
```
datasets                  4.8.5
peft                      0.19.1
accelerate                1.13.0
diffusers                 0.37.1
transformers              5.5.4
```

### Who can help?

@BenjaminBossan  

### Reproduction

```
pytest tests/test_common_gpu.py::PeftGPUCommonTests::test_apply_GS_hra_conv2d_inference
```

### Expected behavior

```
E           FileNotFoundError: [Errno 2] No such file or directory: '/root/.cache/huggingface/hub/datasets--huggingface--cats-image/snapshots/4613f5f1d3642cc2d56ffdf1b58c9d0f912cdc1f/cats_image.jpeg'
```

--- a/tests/test_xlora.py
+++ b/tests/test_xlora.py
@@ -109,7 +109,7 @@ class TestXlora:
 
     @pytest.fixture(scope="class")
     def tokenizer(self):
-        tokenizer = AutoTokenizer.from_pretrained(self.model_id, trust_remote_code=True, device_map=self.torch_device)
+        tokenizer = AutoTokenizer.from_pretrained(self.model_id, device_map=self.torch_device)
         return tokenizer
 
     @pytest.fixture(scope="function")

--- a/tests/testing_utils.py
+++ b/tests/testing_utils.py
@@ -289,7 +289,7 @@ def load_dataset_english_quotes():
 @lru_cache
 def load_cat_image():
     # can't use pytest fixtures for now because of unittest style tests
-    dataset = load_dataset("huggingface/cats-image", trust_remote_code=True)
+    dataset = load_dataset("huggingface/cats-image")
     image = dataset["test"]["image"][0]
     return image
 

